Compile failure on MSVC

MS Visual Studio 2008 (which is used for the default python.org Python 2.7 builds) fails to compile lz4 with the following error:

    src/python-lz4.c(131) : error C2143: syntax error : missing ';' before 'type'

This is because MSVC 2008 doesn't permit mixing declarations with code. The fix is simple, and a patch is forthcoming.

I'm getting a new compile failure with MSVC with the merged refactor branch:

    lz4/frame/_frame.c(175) : error C2059: syntax error : '{'

The problem here is that MSVC doesn't support compound-literal syntax. As a workaround, you could declare `frameInfo` in a local scope with an initializer and assign it to `prefs->frameInfo` then.

--- a/lz4/frame/_frame.c
+++ b/lz4/frame/_frame.c
@@ -153,7 +153,6 @@ static PyObject *
 py_lz4f_makePrefs (PyObject * Py_UNUSED (self), PyObject * args,
 		   PyObject * keywds)
 {
-  LZ4F_frameInfo_t frameInfo;
   LZ4F_preferences_t *prefs;
   PyObject *result = PyDict_New ();
   static char *kwlist[] = { "blockSizeID", "blockMode", "chkFlag"
@@ -171,12 +170,13 @@ py_lz4f_makePrefs (PyObject * Py_UNUSED (self), PyObject * args,
     }
 
   prefs = calloc (1, sizeof (LZ4F_preferences_t));
-  frameInfo = (LZ4F_frameInfo_t)
   {
-    blkID, blkMode, chkSumFlag, 0, 0,
-    {0, 0}
-  };
-  prefs->frameInfo = frameInfo;
+    LZ4F_frameInfo_t frameInfo = {
+      blkID, blkMode, chkSumFlag, 0, 0,
+      {0, 0}
+    };
+    prefs->frameInfo = frameInfo;
+  }
   prefs->autoFlush = autoFlush;
   result = PyCapsule_New (prefs, NULL, NULL);
 

